### PR TITLE
Refactor label name handling for PEFT models in Trainer class

### LINKED ISSUE #3694: No loss logging when using LoraConfig + SFTTrainer
### Reproduction

Hi,

when using LoRA from peft together with SFTTrainer, the evaluation loss is not logged correctly. However, this can be manually activated by setting ```trainer.can_return_loss = True```. So there seems to be a wrong default parameter for this setting.

```python
from transformers import AutoModelForCausalLM
from datasets import load_dataset
from trl import SFTConfig, SFTTrainer
from peft import LoraConfig

dataset = load_dataset("stanfordnlp/imdb", split="train[:500]")
dataset = dataset.train_test_split(test_size=0.01)

model = AutoModelForCausalLM.from_pretrained("facebook/opt-350m")

peft_config = LoraConfig(
    r=4,
    lora_alpha=16,
    lora_dropout=0.05,
    bias="none",
    target_modules="all-linear",
)

training_args = SFTConfig(max_length=2048,
                          output_dir="/tmp",
                          eval_strategy='steps',
                          eval_steps=10,
                          eval_on_start=True,
                          report_to='wandb')

trainer = SFTTrainer(
    model,
    train_dataset=dataset['train'],
    eval_dataset=dataset['test'],
    args=training_args,
    peft_config=peft_config, # comment out for the first setting without Lora
)

# Add the following line to enable logging of evaluation loss
# trainer.can_return_loss = True

trainer.train()
```

Outputs when NOT using LoRA:
```
{'eval_loss': 3.513484239578247, 'eval_runtime': 0.557, 'eval_samples_per_second': 8.977, 'eval_steps_per_second': 1.795, 'eval_num_tokens': 0.0, 'eval_mean_token_accuracy': 0.3457943797111511, 'epoch': 0}                                                                                                                                                             
{'loss': 3.4785, 'grad_norm': 5.905367851257324, 'learning_rate': 1.903225806451613e-05, 'num_tokens': 24041.0, 'mean_token_accuracy': 0.3565803587436676, 'epoch': 0.16}                                                                                                                                                                                                 
```


Outputs when using LoRA:
```
{'eval_runtime': 0.3156, 'eval_samples_per_second': 15.841, 'eval_steps_per_second': 3.168, 'epoch': 0}                                                                                                                                                                                                                                                                   
{'loss': 3.4884, 'grad_norm': 0.7195760011672974, 'learning_rate': 1.903225806451613e-05, 'num_tokens': 24104.0, 'mean_token_accuracy': 0.35992352962493895, 'epoch': 0.16}                                                                                                                                                                                               
```


Outputs when using LoRA + setting trainer.can_return_loss = True :
```
{'eval_loss': 3.567966938018799, 'eval_runtime': 0.2626, 'eval_samples_per_second': 19.043, 'eval_steps_per_second': 3.809, 'eval_num_tokens': 0.0, 'eval_mean_token_accuracy': 0.3465627133846283, 'epoch': 0}                                                                                                                                                           
{'loss': 3.5367, 'grad_norm': 0.838955819606781, 'learning_rate': 1.903225806451613e-05, 'num_tokens': 21051.0, 'mean_token_accuracy': 0.35161270797252653, 'epoch': 0.16}                                                                                                                                                                                                
```


### System Info

- Platform: Linux-5.4.0-198-generic-x86_64-with-glibc2.35
- Python version: 3.11.10
- TRL version: 0.18.1
- PyTorch version: 2.7.0
- accelerator(s): NVIDIA A100-SXM4-80GB, NVIDIA A100-SXM4-80GB, NVIDIA A100-SXM4-80GB, NVIDIA A100-SXM4-80GB, NVIDIA A100-SXM4-80GB, NVIDIA A100-SXM4-80GB, NVIDIA A100-SXM4-80GB, NVIDIA A100-SXM4-80GB
- Transformers version: 4.52.4
- Accelerate version: 1.3.0
- Accelerate config: not found
- Datasets version: 3.6.0
- HF Hub version: 0.33.0
- bitsandbytes version: 0.46.0
- DeepSpeed version: 0.17.0
- Diffusers version: not installed
- Liger-Kernel version: not installed
- LLM-Blender version: not installed
- OpenAI version: 1.91.0
- PEFT version: not installed
- vLLM version: 0.9.0.1

### Checklist

- [x] I have checked that my issue isn't already filed (see [open issues](https://github.com/huggingface/trl/issues?q=is%3Aissue))
- [x] I have included my system information
- [x] Any code provided is minimal, complete, and reproducible ([more on MREs](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any code provided is properly formatted in code blocks, (no screenshot, [more on code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any traceback provided is complete

### PR DESCRIPTION
**Pull request by qgallouedec**
# What does this PR do?

Fix https://github.com/huggingface/trl/issues/3694

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@SunMarc

### CODE REVIEW

**Review by SunMarc — APPROVED** [review]
Makes sense, thanks !
